core/store/src/trie/resharding.rs

```diff
@@ -317,18 +454,63 @@ pub fn get_delayed_receipts(
     Ok(Some((delayed_receipt_indices.first_index, receipts)))
 }
 
+/// Retrieve PromiseYield timeouts starting with `start_index` until `memory_limit` is hit
+/// return None if there is no delayed receipts with index >= start_index
+pub fn get_promise_yield_timeouts(
+    state_update: &TrieUpdate,
+    start_index: Option<u64>,
+    memory_limit: ByteSize,
+) -> Result<Option<(u64, Vec<PromiseYieldTimeout>)>, StorageError> {
```

Comment:
nit: Can you either create a dedicated return type for this (`struct GetPromiseYieldTimeoutsResult { thing: u64, timeouts: Vec<...> }`) or document what's in the return type in the comment?